src/frontend/js/api/joanie.ts

```diff
@@ -191,17 +191,31 @@ export const getRoutes = () => {
 export const isJoanieEnabled = !!context.joanie_backend;
 
 export const buildApiUrl = <ApiFilters extends ResourcesQuery = ResourcesQuery>(
-  url: string,
+  raw_url: string,
   filters?: ApiFilters,
 ) => {
-  const { id = '', ...queryParameters } = filters || {};
-  if (id) url = url.replace(':id', id);
-  else url = url.replace(':id/', '');
-
-  if (!ObjectHelper.isEmpty(queryParameters)) {
-    url += '?' + queryString.stringify(queryParameters);
+  // eslint-disable-next-line compat/compat
+  const url = new URL(raw_url);
+  if (filters) {
+    Object.entries(filters).forEach(([key, value]) => {
+      if (url.pathname.search(`:${key}`) > 0) {
```

Comment:
You could first verify that `value` is not `undefined` at the very start of the callback instead of verifying it two times, WDYT?